src/numeric/impl_numeric.rs

```diff
@@ -283,10 +283,11 @@ where
     ///     a.mean_axis(Axis(0)).unwrap().mean_axis(Axis(0)).unwrap() == aview0(&3.5)
     /// );
     /// ```
-    pub fn mean_axis(&self, axis: Axis) -> Option<Array<A, D::Smaller>>
+    pub fn mean_axis(&self, axis: Axis) -> Option<Array<A, <D::Smaller as BroadcastShape<Ix0>>::BroadcastOutput>>
     where
         A: Clone + Zero + FromPrimitive + Add<Output = A> + Div<Output = A>,
         D: RemoveAxis,
+        D::Smaller: BroadcastShape<Ix0>,
```

Comment:
This is a good point and something I hadn't considered before. Unfortunately, I don't think there's a way to avoid the extra bound without improvements to the language (either specialization or enum-variants-as-types), since the compiler needs to know that `BroadcastShape` is implemented for all possible combinations of dimension types.

Reply:
Let's see how far we get with the blanket implementation for all `Dimension` with output `Self` and then maybe it can be added that `Ix0` can broadcast to any other dimension.

Reply:
I can't think of a good way :( If we add a statement like this:

        impl<D: Dimension> BroadcastShape<Ix0> for D {
            type BroadcastOutput = D;
        }
Then it will conflict with the statement on broadcast.rs:62( Although they don’t actually conflict ).